### PR TITLE
RFC: Improving docs about deploying medusa in production

### PR DESCRIPTION
**Pull request by flashjames**
## What
- Add docs for how to use pm2 and explain why it's needed
  - Possibly explain the other options of process manager
- Possibly: Make a command in package.json medusa-starter-default to start medusa with pm2 with sane defaults.

## Why
- For a person coming to medusajs and putting it in production it's not apparent you need a process manager (especially since there's no mentions in docs about process manager)
  - One common missunderstanding might be that since nodejs has a event loop a process manager is not needed
  - Django has really good docs about how and why you should use Gunicorn as process manager in production, could take inspiration from their docs pages how I would write the docs page,
- It's not apparent how to start medusajs with pm2, you have to go through the whole command code for "medusa develop" to understand which is the entry point file for medusa-starter-default and medusa, which you need to start medusa with pm2

### Route 1
Add a docs page under https://docs.medusajs.com/deployments/server/ explaining why you need a process manager. With the name deploying medusa with pm2.

In each deployment guide it says "change start command" => "medusa migrations run && medusa start"
For example here: "https://docs.medusajs.com/deployments/server/deploying-on-railway#change-start-command
Link to the new https://docs.medusajs.com/deployments/server/deploy-with-pm2

### Route 2
Add pm2 as a dependency to medusa-starter-default and add a new command, for example "yarn start-production": "medusa run migrations && pm2 start index.js"
Link to the new docs page about pm2 in each "deploying on X", explaining it in more detail.

#### Django docs and other posts to take inspiration from
- https://docs.djangoproject.com/en/4.1/howto/deployment/wsgi/gunicorn/
- https://docs.djangoproject.com/en/4.1/howto/deployment/
- https://serverfault.com/questions/331256/why-do-i-need-nginx-and-something-like-gunicorn
- https://vsupalov.com/what-is-gunicorn/

### CODE REVIEW

**Comment by flashjames** [comment]
# Other questions and thoughts

## The entrypoint in medusa-starter-default
Right now before making this PR, I was trying to figure out how to start medusa with pm2 i.e. finding out what's the entry point to "medusa develop" and it was pretty hard to know where the entrypoint js file is for "medusa develop".
Noticed right now, there was a commit yesterday from adrien with an index.js in medusa-starter. So maybe you're already working on, what I'm suggesting in this RFC.
https://github.com/medusajs/medusa-starter-default/commit/b0328836d4d829e58ec289f9b968a3786c5ff389

## Questions about if I did right to add this as a pull request and rfc
Did I go the right route with doing a pull request as an RFC, or would you rather have it posted as an issue or a discussion? My thought process went like this: Since I will see this through if you think it a good idea and also that I wish to get feedback it fits as an RFC. An issue is for yes issues. And discussions, simply don't get that much notice and since it's high traffic and people generally don't put much thought and time before posting. Since I've put thought and time into this, it deserves at least a little feedback since it might help many people deploying medusa in production, especially those that never but a nodejs application into production.

## Note
I will follow this pull request through with all necessary changes. If you'd rather do it internally and or if you think this change is not needed, give me a hint :)

## Context
I'm learning medusajs by building my own store with drop shipping. I have things I've built custom for both backend and storefront, that I think could be made generic and could be very useful to have in core medusa.
Though this is the first time I've been involved and interested in contributing to a open source project and I have a lot of things i need to improve to be valuable in contributing to an open source project.

## The most apparent improvement I need to do
- Get better at communicating and collaborating asynchronously and not give wall of text, that I've done now. I.e. get better at getting my point across in the fewest words possible. It's a way of respecting people's time and helping myself and others be more productive
- Writing cleaner code, that is less error prone and get better at writing tests i.e. spend enough time on getting the solution perfect. I come from a background working as seed/early-stage technical co-founder, i.e. my focus have been on just getting stuff shipped.
- Get better at communicating through chat and being more on point, without wall of text
TLDR: I have lot's of personal improvements to make and just give me feedback if I'm wasting somebody's time

## Pratice and motivation
To get better at this, I need practice and motivation. Motivation I do have, since medusa technology and company is really interesting to me and it aligns with my ambitions for my ecommerce marketplace (starting point is a beauty store built on medusa).

The other part of the motivation comes from the fact that you at medusa seem to be very experienced and have good knowledge, i.e. be very smart and dedicated. I can see that medusa will be the replacement for Shopify gradually and it might even push them at some point to make part of their product open source.

Side note: I did an exit 6 years back and since then I've been pivoting my way to something I don't need funding to get revenue. I.e. I allocate my time as I wish, so I will at times allocate part of my business working hours to contributing to medusa. 

Ambition: Become core-contributor and when I'm at a stage to take in funding for my startup and start hiring people - allocate time for developers where it fits for them to contribute to medusa and it's ecosystem.

**Comment by pevey** [comment]
I'm very much like the suggestion of additional documentation around deployment, perhaps with examples using pm2.  But I personally would rather not have pm2 added to the basic starter.  Fewer decencies are generally better.  There are many ways to deploy and manage the backend, and my setup for instance does not use pm2.

**Comment by riqwan** [comment]
Hey, thanks for the PR! Since v2 brought a lot of architectural changes, we will be closing this PR since it no longer applies to our new setup. cc @shahednasser
